app/models/time_profile.rb

```diff
@@ -37,31 +38,31 @@ def global?
     profile_type == "global"
   end
 
-  def ts_in_profile?(ts, default_tz = DEFAULT_TZ)
-    ts = Time.parse(ts) if ts.kind_of?(String)
-    self.ts_day_in_profile?(ts, default_tz) && self.ts_hour_in_profile?(ts, default_tz)
+  def ts_in_profile?(tstamp, default_tz = DEFAULT_TZ)
+    tstamp = Time.zone.parse(tstamp) if tstamp.kind_of?(String)
+    ts_day_in_profile?(tstamp, default_tz) && ts_hour_in_profile?(tstamp, default_tz)
```

Comment:
I don't really like this one - `ts` is a perfectly fine variable name in this context, IMO

Reply:
Restored.